### ISSUE TITLE
Inconsistencies regarding restore-keys

### ISSUE THREAD

**Issue by afranke** [actions/cache#667]
The README calls for an “ordered list”, which readers will assume means a YAML sequence (sometimes referred to as arrays). [This documentation](https://docs.github.com/en/actions/advanced-guides/caching-dependencies-to-speed-up-workflows#matching-a-cache-key) also mentions “an ordered list”, but the examples use a block scalar (multiline string). The [examples in the repo](https://github.com/actions/cache/blob/main/examples.md) use block scalars as well.

Can you clarify whether a scalar or a sequence is expected? If it is the former, can you rephrase the documentation to something less confusing?

**Comment by ashleypittman** [comment]
I've got a case where I'm using python to inspect the git history and construct a set of restore-keys, and it would be really useful to be able to use a variable number of keys depending on the history but it doesn't look like that's currently possible.

**Comment by bishal-pdMSFT** [comment]
The expectation is for it to be scalar. This is due to limitation with underlying toolkit code which only supports string type input.
This is at par with any other step input in workflow.
